statsmodels/tsa/vector_ar/util.py

```diff
@@ -193,7 +197,7 @@ def varsim(coefs, intercept, sig_u, steps=100, initvalues=None, seed=None):
     p, k, k = coefs.shape
     ugen = rmvnorm(np.zeros(len(sig_u)), sig_u, steps)
     result = np.zeros((steps, k))
-    result[p:] = intercept + ugen[p:]
+    result[p:] = intercept + ugen[p:] if intercept is not None else ugen[p:]
```

Comment:
It isn't immediately obvious whether the order of operations is `(intercept+ugen[p:]) ...` or `intercept + (ugen[p:] ...)`.  In context it isn't hard to figure out, but I still suggest separating out the if/else condition.

This also helps make test coverage measurement more meaningful.